server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/PopulateEmailQueryViewListener.java

```diff
@@ -156,13 +157,22 @@ private Mono<Void> handleAdded(Added added) {
 
     private Mono<Void> handleAdded(Added added, MessageMetaData messageMetaData, MailboxSession session) {
         MessageId messageId = messageMetaData.getMessageId();
+        MailboxId mailboxId = added.getMailboxId();
 
-        return Flux.from(messageIdManager.getMessagesReactive(ImmutableList.of(messageId), FetchGroup.HEADERS, session))
-            .next()
+        return checkMessageStillInOriginMailbox(messageId, session, mailboxId)
+            .filter(FunctionalUtils.identityPredicate())
+            .flatMap(stillInOriginMailbox -> Flux.from(messageIdManager.getMessagesReactive(ImmutableList.of(messageId), FetchGroup.HEADERS, session))
+                .next())
             .filter(message -> !message.getFlags().contains(DELETED))
             .flatMap(messageResult -> handleAdded(added.getMailboxId(), messageResult));
     }
```

Comment:
Please do this check only if `added.getMailboxPath().getName().isEqualTo(OUTBOX)` to limit its cost....